File: Carly_Michael/jectacular-ql/src/app/domain/ast/qls/visitors/qls-visitor.ts
import {QlsQuestion} from '../qls-question';
import {Stylesheet} from '../stylesheet';
import {Page} from '../page';
import {Section} from '../section';

export interface QlsVisitor<T> {
  visitStylesheet(stylesheet: Stylesheet): T;
  visitPage(page: Page): T;
  visitSection(section: Section): T;
  visitQlsQuestion(qlsQuestion: QlsQuestion): T;
}

File: DogeQL/src/main/kotlin/qls/visitor/QlsVisitor.kt
package qls.visitor

import qls.ast.node.*
import qls.ast.node.attribute.Attribute
import qls.ast.node.attribute.DefaultAttributes


interface QlsVisitor<out T> {

    fun visit(styleSheet: StyleSheet): T
    fun visit(page: Page): T
    fun visit(style: Style): T
    fun visit(section: Section): T
    fun visit(defaultAttributes: DefaultAttributes): T
    fun visit(question: Question): T
    fun visit(element: Element): T
    fun visit(attributes: Attribute): T
}